test/entrypoint.test.ts

```diff
@@ -387,6 +387,46 @@ describe('EntryPoint', function () {
     })
   })
 
+  describe('flickering account validation', () => {
+    it('should prevent leakage of basefee', async () => {
+      const maliciousAccount = await new MaliciousAccount__factory(ethersSigner).deploy(entryPoint.address, { value: '0x' + 1e18.toString(16) })
+
+      const snap = await ethers.provider.send('evm_snapshot', [])
+      await ethers.provider.send('evm_mine', [])
+      const block = await ethers.provider.getBlock('latest')
+      await ethers.provider.send('evm_revert', [snap])
+
+      if (block.baseFeePerGas == null) {
+        expect.fail(null, null, 'test error: no basefee')
+      }
+
+      const userOp: UserOperation = {
+        sender: maliciousAccount.address,
+        nonce: '0x0',
+        initCode: '0x',
+        callData: '0x',
+        callGasLimit: '0x' + 1e5.toString(16),
+        verificationGasLimit: '0x' + 1e5.toString(16),
+        preVerificationGas: '0x' + 1e5.toString(16),
+        maxFeePerGas: block.baseFeePerGas.mul(3),
+        maxPriorityFeePerGas: block.baseFeePerGas!,
+        paymasterAndData: '0x',
+        signature: '0x'
+      }
+      try {
+        await expect(entryPoint.simulateValidation(userOp, { gasLimit: 1e6 }))
+          .to.revertedWith('ValidationResult')
+        console.log('after first simulation')
+        await ethers.provider.send('evm_mine', [])
+        await expect(entryPoint.simulateValidation(userOp, { gasLimit: 1e6 }))
+          .to.revertedWith('FailedOp')
+        expect.fail(null, null, 'should fail on first simulation')
+      } catch (e: any) {
+        expect(e.message).to.include('FailedOp')
```

Comment:
I don't think you this catch and "expect" - the first simulation "succeds" (revets with ValidationResult)
the 2nd simulation fails (wit FailedOn) - so this outer catch is not in use

Reply:
What?